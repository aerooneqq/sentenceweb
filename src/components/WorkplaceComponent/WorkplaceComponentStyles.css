#contentContainer{
  width: 100%;
  height: 100%;
  background: white;
  flex-grow: 1;

  box-sizing: border-box;

  display: flex;
  flex-direction: column;
} 

#workplaceContainer{ 
  width: 100%;
  height: 100%;

  box-sizing: border-box;

  overflow-y: scroll;
}

#workplaceContainer::-webkit-scrollbar{ 
  width: 0px;
}
