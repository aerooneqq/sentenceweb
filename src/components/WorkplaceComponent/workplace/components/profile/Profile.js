import React, {Component, lazy} from "react"
import "./ProfileStyles.css"

//Components
import ProfileData from "./components/ProfileData/ProfileData";
import UserPhoto from "./components/UserPhoto/UserPhoto";
import UserActivityComponent from "./components/UserActivity/UserActivity";
import UserFeed from "./components/UserFeed/UserFeed";

/**
 * This component is a top-level component which unites UserPhoto, UserFeed,
 * ProfileData, UserActivity
 * 
 * PROPS LIST:
 * 1) signOut - the function to sign out from the system
 */
export default class Profile extends Component{
  
  constructor(props) {
    super(props);

    this.state = { 
      currentUserDataIndex: 2
    };

    this.changeCurrentUserData = this.changeCurrentUserData.bind(this);
  }

  changeCurrentUserData(newUserDataIndex) {
    this.setState({ 
      currentUserDataIndex: newUserDataIndex
    });
  }

  render() {
    return (
      <div id = "profileContainer">
        <div id = "userPhotoCont">
            <UserPhoto changeUserData = {this.changeCurrentUserData} 
                       signOut = {this.props.signOut}/>
        </div>

        <div id = "rightProfileCont">
          <div id = "userActivitiesOutterCont">
            <UserActivityComponent />
          </div>

          <div id = "userSettingsAndFeedOutterCont">
            <ProfileData currentUserDataIndex = {this.state.currentUserDataIndex} />
            <UserFeed />
          </div>
        </div>
      </div>
    )
  }
}
