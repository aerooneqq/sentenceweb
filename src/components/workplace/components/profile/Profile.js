import React, {Component, Suspense, lazy} from "react"
import "./ProfileStyles.css"

const UserPhoto = lazy(()=>import("./components/UserPhoto"))
const SignOutComponent = lazy(()=>import("./components/userPhotoAtoms/SignOutComponent"))
const DeleteAccountComponent = lazy(()=>import("./components/userPhotoAtoms/DeleteAccountComponent"))
const ProfileDataComponent = lazy(()=>import("./components/ProfileDataComponent"))
const UserActivityComponent = lazy(()=>import("./components/userActivityAtoms/UserActivityComponent"))
const ActivityHeader = lazy(()=>import("./components/userActivityAtoms/ActivityHeader"))


export default class Profile extends Component{
  constructor(){
    super()
  }

  render(){
    return (
      <div id = "profileContainer">
        <div id = "userPhotoCont" className="profileContainer">
          <div id = "innerContainer">
            <UserPhoto />
            <SignOutComponent />
            <DeleteAccountComponent />
          </div>
        </div>

        <div id = "profileDataCont" className="profileContainer">
          <ProfileDataComponent />
        </div>

        <div>
          <ActivityHeader />
          <div id = "userActivityContainer" className="profileContainer">
            <UserActivityComponent />
          </div>
        </div>
      </div>
    )
  }
}
