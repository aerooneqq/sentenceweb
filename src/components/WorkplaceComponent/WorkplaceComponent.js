import React, {Component, lazy, Suspense} from "react"

//Styles
import "./WorkplaceComponentStyles.css"

//Components
const Header = lazy(()=> import("./workplace/components/header/Header"))
const Profile = lazy(()=> import("./workplace/components/profile/Profile"))
const DocumentDesk = lazy(() => import("./workplace/components/workplace/DocumentDesk"));

/**
 * PROPS LIST: 
 * 1) signOut - the method to signout from the system
 */
export default class WorkplaceComponent extends Component{
  constructor(props) { 
    super(props);

    this.state = { 
      component: <DocumentDesk />,

    };

    this._changeWorkplaceComponent = this._changeWorkplaceComponent.bind(this);
  }

  /**
   * Changes the current component of the workplace.
   * @param {*} componentNumber - the number of the component:
   * 0 - Profile
   * 1 - Workplace
   * 2 - Projects
   * 3 - Templates
   */
  _changeWorkplaceComponent(componentNumber){ 
    switch (componentNumber){ 
      case 0:
        this.setState({ component: <Profile signOut = {this.props.signOut}/>});
        break;
      
      case 1:
        this.setState({ component: <DocumentDesk />});
        break;
    }
  }

  render(){
    return (
        <div id = "workplaceContainer">
          <Header changeWorkplaceComponent = {this._changeWorkplaceComponent}/>
          <div id="contentContainer">
            {this.state.component}
          </div>
        </div>
    )
  }
}
