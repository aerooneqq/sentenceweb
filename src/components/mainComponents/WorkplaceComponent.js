import React, {Component, Suspense, lazy} from "react"
import "./styles/WorkplaceComponentStyles.css"

const Header = lazy(()=> import("../workplace/components/header/Header"))
const Profile = lazy(()=> import("../workplace/components/profile/Profile"))

export default class WorkplaceComponent extends Component{
  constructor(props){
    super(props)
  }

  render(){
    return (
      <Suspense fallback = {<div>Loading...</div>}>
        <div>
          <Header />
          <div id="contentContainer">
            <Profile />
          </div>
        </div>
      </Suspense>
    )
  }
}
