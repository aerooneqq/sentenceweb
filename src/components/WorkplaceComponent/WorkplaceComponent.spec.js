import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import WorkplaceComponent from './WorkplaceComponent';
import Profile from "./workplace/components/profile/Profile"

describe("workplace component", () => { 
    it("should render correctly", () => { 
        const props = { 
            signOut: () => {}
        }

        const workplaceComponent = shallow(<WorkplaceComponent />);
        
        expect(shallowToJson(workplaceComponent)).toMatchSnapshot();
    });
});