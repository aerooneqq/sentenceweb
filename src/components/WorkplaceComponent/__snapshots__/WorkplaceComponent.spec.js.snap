// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workplace component should render correctly 1`] = `
<div
  id="workplaceContainer"
>
  <lazy
    changeWorkplaceComponent={[Function]}
  />
  <div
    id="contentContainer"
  >
    <lazy />
  </div>
</div>
`;
